### PR TITLE
Seq2seq ae

### PR DESCRIPTION
**Pull request by stegben**
Usage:

```python
from seq2vec import Seq2SeqAutoEncoderUseWordHash


transformer = Seq2SeqAutoEncoderUseWordHash(
    max_index=10000,
    max_length=5,
    latent_size=20,
)
train_seq = [
    ['我', '有', '一個', '蘋果'],
    ['我', '有', '一支', '筆'],
    ['我', '有', '一個', '鳳梨'],
]
test_seq = [
    ['我', '愛', '吃', '鳳梨'],
]
transformer.fit(train_seq)
result = transformer.transform(test_seq)
# array([[ 0.06332756, -0.02303559, -0.00814635,  0.07087014, -0.05075584,
#         -0.02635699, -0.0356812 , -0.03665418,  0.00562234,  0.02313618,
#         -0.06245217, -0.02793446, -0.02569969, -0.03095961,  0.00557634,
#          0.04762301,  0.028506  ,  0.00861616,  0.00546405, -0.04727045]], dtype=float32)
```

### CODE REVIEW

File: seq2vec/seq2seq_auto_encoder.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        model, encoder = _create_single_layer_seq2seq_model(
+            max_length=self.max_length,
+            max_index=self.max_index,
+            latent_size=self.latent_size,
+        )
+        self.model = model
+        self.encoder = encoder
+
+    def _hash_seq(self, sequence):
+        return [hash(word) % self.max_index for word in sequence]
```
Comment (SoluMilken) [inline seq2vec/seq2seq_auto_encoder.py:49]: hash function to murmur hash

File: seq2vec/seq2seq_auto_encoder.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    def _generate_padding_array(self, seqs):
+        hashed_seq = []
+        for seq in seqs:
+            hashed_seq.append(self._hash_seq(seq))
+        data_pad = pad_sequences(hashed_seq, maxlen=self.max_length)
```
Comment (SoluMilken) [inline seq2vec/seq2seq_auto_encoder.py]: pad_sequences ([0, 0, ..., 0, short sequence], long_sequence[: self.max_length])

File: seq2vec/seq2seq_auto_encoder.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        array = []
+        for seq in data_pad:
+            np_seq = _one_hot_encode_seq(seq, self.max_index)
+            array.append(np_seq)
+        return np.array(array)
+
+    def fit(self, train_seqs):
```
Comment (SoluMilken) [inline seq2vec/seq2seq_auto_encoder.py:67]: the format of train_seqs ?? (numpy array of numpy array ? list of list (inner list can have different length? )

File: seq2vec/seq2seq_auto_encoder.py (hunk 1 above)
Comment (SoluMilken) [reply seq2vec/seq2seq_auto_encoder.py:49]: from sklearn.utils import murmurhash3_32
murmurhash3_32('a', seed=0)
現在有統一seed嗎？？？？

**Review by plliao — APPROVED** [review]
